### ISSUE TITLE
[installation] Failure on intalling using Kubeflow using the GCP UI

### ISSUE THREAD

**Issue by hackintoshrao** [kubeflow/kubeflow#4670]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
I tried installing Kubeflow using the UI based deployment manager https://deploy.kubeflow.cloud/#/deploy. 

But it fails complaining that the couple of GCP APIs are not enabled, but, that's not the case.
<img width="1295" alt="Screenshot 2020-01-21 at 11 29 13 AM" src="https://user-images.githubusercontent.com/2609511/72779566-59c7dd00-3c42-11ea-89e9-cada77b16c8a.png">

I've enabled these APIs for the project used during installation.
<img width="1438" alt="Screenshot 2020-01-21 at 11 29 27 AM" src="https://user-images.githubusercontent.com/2609511/72779570-62b8ae80-3c42-11ea-9c44-8792461eb5bb.png">
<img width="1432" alt="Screenshot 2020-01-21 at 11 29 37 AM" src="https://user-images.githubusercontent.com/2609511/72779571-651b0880-3c42-11ea-8bbd-ae068580ce63.png">

**What did you expect to happen:**
I expected to successfully install Kubeflow

**Environment:**

- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard): v0.7
- kfctl version: (use `kfctl version`): NA
- Kubernetes platform: GKE (installation aken care by the UI)
- Kubernetes version: (use `kubectl version`): NA
- OS (e.g. from `/etc/os-release`): NA

**Comment by hackintoshrao** [comment]
This issue exists both while installing using the GCP cloud IAP and the basic auth.

**Comment by jlewi** [comment]
@hackintoshrao could you please run 

```
gcloud --project=${PROJECT} services list
```

So we can see which services are enabled for your project?

Could you then try

1. Running

    ```
    gcloud services enable \
  cloudresourcemanager.googleapis.com \
  iam.googleapis.com \
  file.googleapis.com \
  ml.googleapis.com
    ```

1. Rerunning
  
    ```
    gcloud --project=${PROJECT}  services list
    ```
1. Retrying with the UI to see if that fixes the issue

@hackintoshrao If that fails you could try using the CLI as a workaround.

@kunmingg this seems similar to issues we had seen in the past with not enabling certain APIs.

**Comment by kunmingg** [comment]
Hi @hackintoshrao,
Can you add the logs from kubeflow UI?

**Comment by hackintoshrao** [comment]
Hey @kunmingg ,

Here is the log from the Kubeflow UI 

```
2020-00-3119:06:46.678:Gettingenabledservicesforprojectkubeflow-testing-266813..

2020-00-3119:07:02.306:CouldnotconfigurecommunicationwithGCP,exiting

2020-00-3119:07:44.775:Gettingenabledservicesforprojectkubeflow-testing-266813..

2020-00-3119:07:57.958:CouldnotconfigurecommunicationwithGCP,exiting
```

Besides the above error, the following popup shows up.

<img width="1240" alt="Screenshot 2020-01-31 at 7 11 29 PM" src="https://user-images.githubusercontent.com/2609511/73543694-8ad3b900-445d-11ea-81bf-67e475c18242.png">

Here is the output of `gcloud --project=${PROJECT} services list`, I've got the required services enabled, billing account setup and I'm not on the free tier, I also used a private browser window during my attempt to install Kubeflow using the deployment UI.

This deployment is helpful as we are working on getting started videos series for Kubeflow.

**Comment by jlewi** [comment]
Looks like the error is coming from here:
https://github.com/kubeflow/kubeflow/blob/02f293c74d41f4240bfc5ddf2181b23ae7c1179c/components/gcp-click-to-deploy/src/DeployForm.tsx#L695

It looks like the code is never able to successfully get the project number
https://github.com/kubeflow/kubeflow/blob/02f293c74d41f4240bfc5ddf2181b23ae7c1179c/components/gcp-click-to-deploy/src/DeployForm.tsx#L928

@hackintoshrao Are you able to run ?

```
gcloud projects describe ${PROJECT}
```

Can you also run `gcloud auth list`? Does the current account match the account you are logging into the UI with?

**Comment by kunmingg** [comment]
Are you log in as owner account of the project?
And the UI expect project id (not project name) to specify target project.

**Comment by hackintoshrao** [comment]
@kunmingg : Yes, I've logged in as the owner and got the Project ID right. I would expect the UI provide me an useful error which I can act upon.

**Comment by jlewi** [comment]
Looks like we got another report of this; #4919. Are we swallowing the error here?
https://github.com/kubeflow/kubeflow/blob/02f293c74d41f4240bfc5ddf2181b23ae7c1179c/components/gcp-click-to-deploy/src/DeployForm.tsx#L695

@kunmingg should we fix the logging to print out the actual error?

**Comment by kunmingg** [comment]
https://github.com/kubeflow/kubeflow/issues/4921

**Comment by BioGeek** [comment]
Repeating my comment from #4919: 
I think I realized what I did wrong in the UI: I used the project name instead of the project ID.

**Comment by polsm91** [comment]
Hi, I encountered the same issue as @hackintoshrao . I made the same checks and also that the project ID was correct. Then, I tried with the CLI using the same parameters, and it worked.

I deleted the deployment and tried again through the [UI](https://deploy.kubeflow.cloud) but it kept failing.

**Comment by jlewi** [comment]
@polsm91 Can you try the CLI?

**Comment by polsm91** [comment]
@jlewi The CLI was working just fine; my issues were related to the UI deployment.

I just tried to deploy Kubeflow using the web UI and the same IAP credentials as last time, and now it works.

The first time I tried the GCE account was new and followed the [tutorial](https://www.kubeflow.org/docs/gke/gcp-e2e/). During these 3 weeks I have enabled other APIs and changed configurations too.

**Comment by jlewi** [comment]
Thanks @polsm91 

We are most likely going to deprecate and remove deploy.kubeflow.cloud. So I don't think we have any plans to fix it.

We added a deprecation notice in kubeflow/kubeflow#4967 it looks like this hasn't been pushed yet.

**Comment by jlewi** [comment]
I updated C2D so the deprecation notice should now show up.
